### PR TITLE
Chore: Refactor code for preventing code duplications

### PR DESCRIPTION
**Pull request by huyhuynh3103**
## Motivation
While deep diving into how `veify-bytecode` works, I found some code duplications and minor typo errors. 

## Solution

Eliminates these duplications and correct the typos, ensuring not to alter its core logic.

### CODE REVIEW

**Review by DaniPopes — APPROVED** [review]
Thanks!

File: crates/verify/src/bytecode.rs [hunk 1]
```diff
@@ -199,10 +199,13 @@ impl VerifyBytecodeArgs {
                 creation_data.transaction_hash
             );
         };
+
         // Extract creation code
-        let maybe_creation_code = if receipt.contract_address == Some(self.address) {
+        let maybe_creation_code = if receipt.to == Some(Address::ZERO) &&
```
Comment (DaniPopes) [inline crates/verify/src/bytecode.rs:204]: shouldn't this also include `receipt.to.is_none()`?

Comment (huyhuynh3103) [reply crates/verify/src/bytecode.rs]: thanks

**Comment by DaniPopes** [comment]
Pending @yash-atreya

**Comment by huyhuynh3103** [comment]
Any updates on here @DaniPopes @yash-atreya

**Review by yash-atreya — APPROVED** [review]
Much cleaner now. Thanks.

**Pull request merged**